# all joint states
joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 50

# left track
track_left_joint_controller:
    type: effort_controllers/JointPositionController
    joint: track_left_joint
    pid: { p: 500.0, i: 0.0, d: 1.0 }

# right track
track_right_joint_controller:
    type: effort_controllers/JointPositionController
    joint: track_right_joint
    pid: { p: 500.0, i: 0.0, d: 1.0 }

# torso
torso_joint1_controller:
    type: effort_controllers/JointPositionController
    joint: torso_joint1
    pid: { p: 4000.0, d: 0.0, i: 0.0, i_clamp: 1.0 }

#torso_joint1_controller:
#    type: effort_controllers/JointPositionController
#    joint: torso_joint1
#    pid: { p: 10000000.0, d: 1000000.0, i: 0.0, i_clamp: 1.0 }

# left limb
#left_limb:
#    type: effort_controllers/JointTrajectoryController
#    joints:
#        - limb_left_joint1
#        - limb_left_joint2
#        - limb_left_joint3
#        - limb_left_joint4
#        - limb_left_joint5
#        - limb_left_joint6
#        - limb_left_joint7
#    gains:
#        limb_left_joint1: { p: 2000.0, i: 0.0, d: 0.0, i_clamp: 1.0 }
#        limb_left_joint2: { p: 2000.0, i: 0.0, d: 0.0, i_clamp: 1.0 }
#        limb_left_joint3: { p: 2000.0, i: 0.0, d: 0.0, i_clamp: 1.0 }
#        limb_left_joint4: { p: 2000.0, i: 0.0, d: 0.0, i_clamp: 1.0 }
#        limb_left_joint5: { p: 2000.0, i: 0.0, d: 0.0, i_clamp: 1.0 }
#        limb_left_joint6: { p: 2000.0, i: 0.0, d: 0.0, i_clamp: 1.0 }
#        limb_left_joint7: { p: 2000.0, i: 0.0, d: 0.0, i_clamp: 1.0 }

# right limb
#right_limb:
#    type: effort_controllers/JointTrajectoryController
#    joints:
#        - limb_right_joint1
#        - limb_right_joint2
#        - limb_right_joint3
#        - limb_right_joint4
#        - limb_right_joint5
#        - limb_right_joint6
#        - limb_right_joint7
#    gains:
#        limb_right_joint1: { p: 2000.0, i: 0.0, d: 1.0, i_clamp: 1.0 }
#        limb_right_joint2: { p: 2000.0, i: 0.0, d: 1.0, i_clamp: 1.0 }
#        limb_right_joint3: { p: 2000.0, i: 0.0, d: 1.0, i_clamp: 1.0 }
#        limb_right_joint4: { p: 2000.0, i: 0.0, d: 1.0, i_clamp: 1.0 }
#        limb_right_joint5: { p: 2000.0, i: 0.0, d: 1.0, i_clamp: 1.0 }
#        limb_right_joint6: { p: 2000.0, i: 0.0, d: 1.0, i_clamp: 1.0 }
#        limb_right_joint7: { p: 2000.0, i: 0.0, d: 1.0, i_clamp: 1.0 }
#    constraints:
#        goal_time: 10.0

#right_limb:
#    type: effort_controllers/JointTrajectoryController
#    joints:
#        - limb_right_joint1
#        - limb_right_joint2
#        - limb_right_joint3
#        - limb_right_joint4
#        - limb_right_joint5
#        - limb_right_joint6
#        - limb_right_joint7
#    gains:
#        limb_right_joint1: { p: 2400.0, d: 18.0, i: 800.0, i_clamp: 4.0 }
#        limb_right_joint2: { p: 1200.0, d: 10.0, i: 700.0, i_clamp: 4.0 }
#        limb_right_joint3: { p: 1000.0, d: 6.0,  i: 600.0, i_clamp: 4.0 }
#        limb_right_joint4: { p: 700.0,  d: 4.0,  i: 450.0, i_clamp: 4.0 }
#        limb_right_joint5: { p: 300.0,  d: 6.0,  i: 300.0, i_clamp: 2.0 }
#        limb_right_joint6: { p: 300.0,  d: 4.0,  i: 300.0, i_clamp: 2.0 }
#        limb_right_joint7: { p: 300.0,  d: 4.0,  i: 300.0, i_clamp: 2.0 }
#    joint_trajectory_action_node:
#        joints:
#            - limb_right_joint1
#            - limb_right_joint2
#            - limb_right_joint3
#            - limb_right_joint4
#            - limb_right_joint5
#            - limb_right_joint6
#            - limb_right_joint7
#        constraints:
#            goal_time: 0.6
#            limb_right_joint1:
#                goal: 0.02
#            limb_right_joint2:
#                goal: 0.02
#            limb_right_joint3:
#                goal: 0.02
#            limb_right_joint4:
#                goal: 0.02
#            limb_right_joint5:
#                goal: 0.02
#            limb_right_joint6:
#                goal: 0.02
#            limb_right_joint7:
#                goal: 0.02

right_limb:
    type: effort_controllers/JointTrajectoryController
    joints:
        - torso_joint1
        - limb_right_joint1
        - limb_right_joint2
        - limb_right_joint3
        - limb_right_joint4
        - limb_right_joint5
        - limb_right_joint6
        - limb_right_joint7
    gains:
#        torso_joint1:      { p: 5000.0, d:  100.0, i:  10.0, i_clamp: 1.0 }
#        limb_right_joint1: { p: 2400.0, d: 18.0, i: 800.0, i_clamp: 4.0 }
#        limb_right_joint2: { p: 1200.0, d: 10.0, i: 700.0, i_clamp: 4.0 }
#        limb_right_joint3: { p: 1000.0, d:  6.0, i: 600.0, i_clamp: 4.0 }
#        limb_right_joint4: { p:  700.0, d:  4.0, i: 450.0, i_clamp: 4.0 }
#        limb_right_joint5: { p:  300.0, d:  6.0, i: 300.0, i_clamp: 2.0 }
#        limb_right_joint6: { p:  300.0, d:  4.0, i: 300.0, i_clamp: 2.0 }
#        limb_right_joint7: { p:  300.0, d:  4.0, i: 300.0, i_clamp: 2.0 }
        torso_joint1:      { p: 20000.0, d:  0.0, i:   0.0, i_clamp: 0.0 }
        limb_right_joint1: { p: 30000.0, d: 0.0, i: 0.0, i_clamp: 0.0 }
        limb_right_joint2: { p: 30000.0, d: 0.0, i: 0.0, i_clamp: 0.0 }
        limb_right_joint3: { p: 30000.0, d: 0.0, i: 0.0, i_clamp: 0.0 }
        limb_right_joint4: { p: 20000.0, d: 0.0, i: 0.0, i_clamp: 4.0 }
        limb_right_joint5: { p: 20000.0, d: 0.0, i: 0.0, i_clamp: 0.0 }
        limb_right_joint6: { p: 20000.0, d: 0.0, i: 0.0, i_clamp: 0.0 }
        limb_right_joint7: { p: 20000.0, d: 0.0, i: 0.0, i_clamp: 0.0 }
    constraints:
        goal_time: 10.0 #0.06
        stopped_velocity_tolerance: 0.01 # default 0.01
        torso_joint1:       { goal: 0.0349 } #, trajectory: 0.0872 } # { goal: 0.02 }
        limb_right_joint1:  { goal: 0.0349 } #, trajectory: 0.0872 } # { goal: 0.02 }
        limb_right_joint2:  { goal: 0.0349 } #, trajectory: 0.0872 } # { goal: 0.02 }
        limb_right_joint3:  { goal: 0.0349 } #, trajectory: 0.0872 } # { goal: 0.02 }
        limb_right_joint4:  { goal: 0.0349 } #, trajectory: 0.0872 } # { goal: 0.02 }
        limb_right_joint5:  { goal: 0.0349 } #, trajectory: 0.0872 } # { goal: 0.02 }
        limb_right_joint6:  { goal: 0.0349 } #, trajectory: 0.0872 } # { goal: 0.02 }
        limb_right_joint7:  { goal: 0.0349 } #, trajectory: 0.0872 } # { goal: 0.02 }
    stop_trajectory_duration: 0.0 # 0.0 disabled

left_limb:
    type: effort_controllers/JointTrajectoryController
    joints:
        - limb_left_joint1
        - limb_left_joint2
        - limb_left_joint3
        - limb_left_joint4
        - limb_left_joint5
        - limb_left_joint6
        - limb_left_joint7
    gains:
#        limb_left_joint1: { p: 2400.0, d: 18.0, i: 800.0, i_clamp: 4.0 }
#        limb_left_joint2: { p: 1200.0, d: 10.0, i: 700.0, i_clamp: 4.0 }
#        limb_left_joint3: { p: 1000.0, d: 6.0,  i: 600.0, i_clamp: 4.0 }
#        limb_left_joint4: { p: 700.0,  d: 4.0,  i: 450.0, i_clamp: 4.0 }
#        limb_left_joint5: { p: 300.0,  d: 6.0,  i: 300.0, i_clamp: 2.0 }
#        limb_left_joint6: { p: 300.0,  d: 4.0,  i: 300.0, i_clamp: 2.0 }
#        limb_left_joint7: { p: 300.0,  d: 4.0,  i: 300.0, i_clamp: 2.0 }
        limb_left_joint1: { p: 30000.0, d: 0.0, i: 0.0, i_clamp: 0.0 }
        limb_left_joint2: { p: 30000.0, d: 0.0, i: 0.0, i_clamp: 0.0 }
        limb_left_joint3: { p: 30000.0, d: 0.0, i: 0.0, i_clamp: 0.0 }
        limb_left_joint4: { p: 20000.0, d: 0.0, i: 0.0, i_clamp: 4.0 }
        limb_left_joint5: { p: 20000.0, d: 0.0, i: 0.0, i_clamp: 0.0 }
        limb_left_joint6: { p: 20000.0, d: 0.0, i: 0.0, i_clamp: 0.0 }
        limb_left_joint7: { p: 20000.0, d: 0.0, i: 0.0, i_clamp: 0.0 }
    constraints:
        goal_time: 10.0 #0.06
        stopped_velocity_tolerance: 0.01 # default 0.01
        torso_joint1:       { goal: 0.0349 } #, trajectory: 0.0872 } # { goal: 0.02 }
        limb_right_joint1:  { goal: 0.0349 } #, trajectory: 0.0872 } # { goal: 0.02 }
        limb_right_joint2:  { goal: 0.0349 } #, trajectory: 0.0872 } # { goal: 0.02 }
        limb_right_joint3:  { goal: 0.0349 } #, trajectory: 0.0872 } # { goal: 0.02 }
        limb_right_joint4:  { goal: 0.0349 } #, trajectory: 0.0872 } # { goal: 0.02 }
        limb_right_joint5:  { goal: 0.0349 } #, trajectory: 0.0872 } # { goal: 0.02 }
        limb_right_joint6:  { goal: 0.0349 } #, trajectory: 0.0872 } # { goal: 0.02 }
        limb_right_joint7:  { goal: 0.0349 } #, trajectory: 0.0872 } # { goal: 0.02 }
    stop_trajectory_duration: 0.0 # 0.0 disabled

right_gripper:
    type: effort_controllers/JointTrajectoryController
    joints:
        - limb_right_finger_1_joint_1
        - limb_right_finger_1_joint_2
        - limb_right_finger_1_joint_3
        - limb_right_finger_2_joint_1
        - limb_right_finger_2_joint_2
        - limb_right_finger_2_joint_3
        - limb_right_finger_middle_joint_1
        - limb_right_finger_middle_joint_2
        - limb_right_finger_middle_joint_3
        - limb_right_palm_finger_1_joint
        - limb_right_palm_finger_2_joint
    gains:
        limb_right_finger_1_joint_1: { p: 100.0, d: 0.0, i: 0.0 }
        limb_right_finger_1_joint_2: { p: 100.0, d: 0.0, i: 0.0 }
        limb_right_finger_1_joint_3: { p: 100.0, d: 0.0, i: 0.0 }
        limb_right_finger_2_joint_1: { p: 100.0, d: 0.0, i: 0.0 }
        limb_right_finger_2_joint_2: { p: 100.0, d: 0.0, i: 0.0 }
        limb_right_finger_2_joint_3: { p: 100.0, d: 0.0, i: 0.0 }
        limb_right_finger_middle_joint_1: { p: 100.0, d: 0.0, i: 0.0 }
        limb_right_finger_middle_joint_2: { p: 100.0, d: 0.0, i: 0.0 }
        limb_right_finger_middle_joint_3: { p: 100.0, d: 0.0, i: 0.0 }
        limb_right_palm_finger_1_joint: { p: 100.0, d: 0.0, i: 0.0 }
        limb_right_palm_finger_2_joint: { p: 100.0, d: 0.0, i: 0.0 }

left_gripper:
    type: effort_controllers/JointTrajectoryController
    joints:
        - limb_left_finger_1_joint_1
        - limb_left_finger_1_joint_2
        - limb_left_finger_1_joint_3
        - limb_left_finger_2_joint_1
        - limb_left_finger_2_joint_2
        - limb_left_finger_2_joint_3
        - limb_left_finger_middle_joint_1
        - limb_left_finger_middle_joint_2
        - limb_left_finger_middle_joint_3
        - limb_left_palm_finger_1_joint
        - limb_left_palm_finger_2_joint
    gains:
        limb_left_finger_1_joint_1: { p: 100.0, d: 0.0, i: 0.0 }
        limb_left_finger_1_joint_2: { p: 100.0, d: 0.0, i: 0.0 }
        limb_left_finger_1_joint_3: { p: 100.0, d: 0.0, i: 0.0 }
        limb_left_finger_2_joint_1: { p: 100.0, d: 0.0, i: 0.0 }
        limb_left_finger_2_joint_2: { p: 100.0, d: 0.0, i: 0.0 }
        limb_left_finger_2_joint_3: { p: 100.0, d: 0.0, i: 0.0 }
        limb_left_finger_middle_joint_1: { p: 100.0, d: 0.0, i: 0.0 }
        limb_left_finger_middle_joint_2: { p: 100.0, d: 0.0, i: 0.0 }
        limb_left_finger_middle_joint_3: { p: 100.0, d: 0.0, i: 0.0 }
        limb_left_palm_finger_1_joint: { p: 100.0, d: 0.0, i: 0.0 }
        limb_left_palm_finger_2_joint: { p: 100.0, d: 0.0, i: 0.0 }
